### PR TITLE
fix: feat/agent api route

### LINKED ISSUE #65: [TEST] Unit Testing: agent-specific issue endpoint
# Test Task: agent-specific issue data

## Goal
Add tests for `agent.js` to ensure correct behavior and edge case handling.

---

## Test Type
- [x] Unit (logic / handler)
- [ ] E2E (full request flow)

---

## Done When
- All tests pass in CI or locally if pipeline testing not setup yet
- Core flows + edge cases covered
- No failing lint/tests

### PR DESCRIPTION
**Pull request by michaelawad3**
Replaces the incorrect agent endpoint with the correct implementation: `GET /agents/:id` to fetch an issue, `POST /agents` for the agent to create issues from the CLI, and `PATCH /agents/:id` to update agent-relevant fields while blocking human-reserved ones. Also removes the `agent_attempts` table from the schema which was referencing a nonexistent `agents` table and breaking all tests. 35 new unit and integration tests added covering all routes.

closes #65

### CODE REVIEW

File: issue-tracker-api/routes/agent.js [hunk 1]
```diff
@@ -1,142 +1,264 @@
...
-	const isValidId = (value) => /^\d+$/.test(value) && Number(value) > 0;
-	const allowedFields = ['name', 'type', 'token'];
+	// -----------------------------------------------------------------------
+	// GET /agents/:id
+	// Fetch a single issue by id. Returns all fields as JSON so the agent has
+	// full context before starting work (stack trace, steps to reproduce, etc.).
+	// Array fields (tags, stack_trace, affected_files) are parsed from their
+	// stored JSON strings back into arrays.
+	// -----------------------------------------------------------------------
+	if (method === 'GET' && issueId) {
```
Comment (Amormio25) [inline issue-tracker-api/routes/agent.js:64]: @michaelawad3 look at either Jonathan or Ben's endpoints for issues or invites for help on this note: we need to make sure user is authenticated before these paths are accessed with requireAuth

File: issue-tracker-api/routes/agent.js [hunk 2]
```diff
@@ -1,142 +1,264 @@
...
+		// Parse JSON array fields back into arrays for the agent
+		const formatted = {
+			...issue,
+			tags: JSON.parse(issue.tags || '[]'),
+			stack_trace: JSON.parse(issue.stack_trace || '[]'),
+			affected_files: JSON.parse(issue.affected_files || '[]'),
+		};
 
-		return Response.json(results);
+		return Response.json(formatted);
```
Comment (Amormio25) [inline issue-tracker-api/routes/agent.js]: 200 response

File: issue-tracker-api/routes/agent.js [hunk 3]
```diff
@@ -1,142 +1,264 @@
...
+	// -----------------------------------------------------------------------
+	// POST /agents
+	// Agent creates a new issue directly from the CLI into the issues table,
+	// so it shows up on the frontend like any human-created issue. Unlike the
+	// human endpoint where the LLM fills in details, the agent populates all
+	// fields itself with full structured context.
+	// -----------------------------------------------------------------------
+	if (method === 'POST' && !issueId) {
+		// gets created_by from session token
+		const header = request.headers.get('Authorization');
```
Comment (Amormio25) [inline issue-tracker-api/routes/agent.js]: same idea here, use the require auth setup we have in other endpoints cuz it already handles the response, also you need to make sure that the caller is a team member, so requireAuth and requireTeamMember

File: issue-tracker-api/routes/agent.js [hunk 4]
```diff
@@ -1,142 +1,264 @@
...
-	if (request.method === 'DELETE' && agentId) {
-		const result = await env.DB.prepare('DELETE FROM agents WHERE id = ?').bind(agentId).run();
-
-		if (result.meta.changes === 0) {
-			return new Response('Agent not found', { status: 404 });
+			return Response.json({ error: 'Issue not found' }, { status: 404 });
 		}
 
-		return new Response('Agent deleted', { status: 200 });
+		return Response.json({ success: true });
```
Comment (Amormio25) [inline issue-tracker-api/routes/agent.js:274]: successfule delete should be 204

File: issue-tracker-api/routes/agent.js [hunk 5]
```diff
@@ -1,142 +1,264 @@
...
-			body = await request.json();
-		} catch (_error) {
-			return new Response('Invalid JSON request body', { status: 400 });
+	// -----------------------------------------------------------------------
+	// PATCH /agents/:id
+	// Agent updates issue fields as it works on a fix. The agent can update
+	// any field EXCEPT immutable record metadata and fields reserved for
+	// human input (description, assigned_to). updated_at is always refreshed.
+	// -----------------------------------------------------------------------
+	if (method === 'PATCH' && issueId) {
```
Comment (Amormio25) [inline issue-tracker-api/routes/agent.js:201]: same here make sure we're authenticated and a team member like:
	const auth = await requireAuth(request, env);
		if (auth.error) return auth.error;
......

const membership = await requireTeamMember(env, auth.userId, issue.team_id);
		if (membership.error) return membership.error;

**Review by Amormio25 — APPROVED** [review]
